### PR TITLE
:bug: Fixed AssetHelper not working with svg

### LINKED ISSUE #13971: AssetHelper not working with svg
### Issue Summary

The asset helper does not work with svg images using # because it adds a query param.

### Steps to Reproduce

```
 <svg class="icon icon__primary quote__line">
        <use xlink:href="{{asset "images/solid.svg#minus"}}"></use>
</svg>
```

Using this does not show the svg images it renders to:
xlink:href="/assets/images/solid.svg#minus?v=094ca0c660"

When I remove the param it works fine and also of cource if  I don't use the helper but this raises a gscan error

### Ghost Version

4.32.0

### Node.js Version

14.18.2

### How did you install Ghost?

Docker Image

### Database type

SQLite3

### Browser & OS version

Chrome / safari

### Relevant log / error output

_No response_

### Code of Conduct

- [X] I agree to be friendly and polite to people in this repository

### PR DESCRIPTION
**Pull request by torstenzander**
closes #13971

This fixes an issue with svg sprites using the # to select an icon, when a ?v=hash is rendered the svg images is not shown. The bugfix removes ?v=hash param for svg files with an #.

Got some code for us? Awesome 🎊!

Please include a description of your change & check your PR against this list, thanks!

- [x] There's a clear use-case for this code change
- [x] Commit message has a short title & references relevant issues
- [x] The build will pass (run `yarn test:all` and `yarn lint`)

More info can be found by clicking the "guidelines for contributing" link above.

### CODE REVIEW

**Comment by ErisDS** [comment]
Hey @torstenzander thanks for the PR :) 

I don't think this is the fix that you want. The `?v=` query param is for cache busting, so this change would mean any svg with a # would get stuck in cache indefinitely.

The correct fix would be to insert the `?v=` param before the #, and maintain both.

Also, this bug would affect absolutely any path with a #, although it currently only really makes sense to use with an SVG, there's no need to limit the solution if implemented this way  🙂

File: test/unit/frontend/meta/asset-url.test.js [hunk 1]
```diff
@@ -34,6 +34,11 @@ describe('getAssetUrl', function () {
         testUrl.should.equal('/public/myfile.js?v=' + config.get('assetHash'));
     });
 
+    it('should return no hash when svg#', function () {
+        const testUrl = getAssetUrl('myfile.svg#arrow-up');
+        testUrl.should.equal('/assets/myfile.svg#arrow-up');
```
Comment (ErisDS) [inline test/unit/frontend/meta/asset-url.test.js:39]:
```
testUrl.should.equal(`/assets/myfile.svg?v=${config.get('assetHash')}#arrow-up`);
```

I'd expect the test to look like this :)

**Review by ErisDS — CHANGES_REQUESTED** [review]
Hey @torstenzander thanks for the PR :)

I don't think this is the fix that you want. The ?v= query param is for cache busting, so this change would mean any svg with a # would get stuck in cache indefinitely.

The correct fix would be to insert the ?v= param before the #, and maintain both.

Also, this bug would affect absolutely any path with a #, although it currently only really makes sense to use with an SVG, there's no need to limit the solution if implemented this way 🙂

**Comment by torstenzander** [comment]
I am not sure if I have to do anything, my tests cover everything I added even though the coverage report looks different.

**Review by ErisDS — APPROVED** [review]
Hey @torstenzander no you don't need to do anything. Thanks for your work :)

**Pull request merged**